src/data.jl

```diff
@@ -196,6 +196,8 @@ struct OccupiedStates
     end
 end
 
+num_states(x::OccupiedStates) = size(x.coeff)[2]
```

Comment:
I think it's generally better to use `size(x.coeff, 2)` rather than indexing. Don't worry about this one as I'm going to alter the `OccupiedStates` struct and I expect some of the code to change anyway.